lockbox-ios/Presenter/FxAPresenter.swift

```diff
@@ -52,7 +52,15 @@ class FxAPresenter {
 extension FxAPresenter {
     func matchingRedirectURLReceived(_ navigationURL: URL) {
         self.dispatcher.dispatch(action: OnboardingStatusAction(onboardingInProgress: true))
-        self.dispatcher.dispatch(action: LoginRouteAction.onboardingConfirmation)
+        self.dispatcher.dispatch(action: self.getOnboardingRoute())
         self.dispatcher.dispatch(action: AccountAction.oauthRedirect(url: navigationURL))
     }
+
+    private func getOnboardingRoute() -> LoginRouteAction {
```

Comment:
nice!

Reply:
I think the other piece here is checking whether autofill is already enabled or not.. if the user has already enabled autofill, then we shouldn't display the autofill onboarding screen.

Reply:
@sashei good catch! Because getting `CredentialProviderStore.state` is an observable should there be a `Observable<LoginRouteAction>` property that holds the state of what action to send?

Reply:
Hm, I suppose it has to be async in some way... maybe have a function called `beginOnboarding()` and that can `.take(1)` from the `CredentialProviderStore.state` to dispatch the appropriate onboarding route action?

I realize looking at the `CredentialProviderStore` that there is no state for `.Allowed` so you might need to add that.